File: HTML/js/settingsService.js
/**
 * 設定服務模組 - 用於管理應用程式設定的讀取和儲存
 */
class SettingsService {
    constructor() {
        this.configKey = 'screenAgent_settings';
        this.defaultSettings = {
            apiKey: '',
            showConversationHistory: true,
            captureFrequencyInSeconds: 1,
            textToSpeechVoiceName: '',
            geminiModel: 'gemini-2.0-flash-exp',
            enableTextToSpeech: true,
            systemPrompt: `你是一個智慧回覆助手，能夠分析桌面截圖並回答使用者的提問。你的回應方式遵循以下規則：
與圖片相關的問題：
仔細分析圖片內容，根據使用者的提問提供準確、清楚的指導與回答。
若問題涉及技術支援，提供具體步驟或建議。
若問題模糊，請要求使用者提供更多資訊，以確保回答準確。
沒有圖片或與圖片無關的問題：
以簡短、直接的方式進行回應，保持自然的對話風格。
若問題超出你的知識範圍，可使用工具查詢最新資訊後回答。
回覆風格：
內容簡潔明瞭，避免冗長或不必要的細節。
禁止使用表情符號、特殊符號或非文字內容。
你應該始終確保回應的專業性與實用性，並根據問題的性質適當調整回覆方式，一律以繁體中文回覆。`
        };
        
        this.settings = this.loadSettings();
        this.autoSave = true;
    }

    /**
     * 從 localStorage 讀取設定
     * @returns {Object} 設定物件
     */
    loadSettings() {
        try {
            // 嘗試從 localStorage 讀取設定
            const savedSettings = localStorage.getItem(this.configKey);
            
            if (savedSettings) {
                // 解析儲存的 JSON 字串為物件
                return JSON.parse(savedSettings);
            }
            
            // 如果沒有找到設定，返回預設設定
            return {...this.defaultSettings};
        } catch (error) {
            console.error('無法讀取設定：', error);
            // 發生錯誤時，返回預設設定
            return {...this.defaultSettings};
        }
    }

    /**
     * 儲存設定到 localStorage
     */
    saveSettings() {
        try {
            // 將設定物件轉換為 JSON 字串，並儲存到 localStorage
            localStorage.setItem(this.configKey, JSON.stringify(this.settings));
        } catch (error) {
            console.error('儲存設定時發生錯誤：', error);
        }
    }

    /**
     * 重置所有設定為預設值
     */
    resetSettings() {
        this.settings = {...this.defaultSettings};
        this.saveSettings();
    }

    // 取得所有設定
    getAllSettings() {
        return {...this.settings};
    }

    // 設定 API 金鑰
    getApiKey() {
        return this.settings.apiKey || '';
    }

    setApiKey(key) {
        this.settings.apiKey = key;
        if (this.autoSave) this.saveSettings();
    }

    // 設定是否顯示對話歷史
    getShowConversationHistory() {
        return this.settings.showConversationHistory !== false;
    }

    setShowConversationHistory(show) {
        this.settings.showConversationHistory = show;
        if (this.autoSave) this.saveSettings();
    }

    // 設定擷取頻率
    getCaptureFrequencyInMilliseconds() {
        return (this.settings.captureFrequencyInSeconds || 1) * 1000;
    }

    setCaptureFrequencyInSeconds(seconds) {
        this.settings.captureFrequencyInSeconds = seconds;
        if (this.autoSave) this.saveSettings();
    }

    // 設定語音名稱
    getTextToSpeechVoiceName() {
        return this.settings.textToSpeechVoiceName || '';
    }

    setTextToSpeechVoiceName(name) {
        this.settings.textToSpeechVoiceName = name;
        if (this.autoSave) this.saveSettings();
    }

    // 設定 Gemini 模型
    getGeminiModel() {
        return this.settings.geminiModel || 'gemini-2.0-flash-exp';
    }

    setGeminiModel(model) {
        this.settings.geminiModel = model;
        if (this.autoSave) this.saveSettings();
    }

    // 設定是否啟用文字轉語音
    getEnableTextToSpeech() {
        return this.settings.enableTextToSpeech !== false;
    }

    setEnableTextToSpeech(enable) {
        this.settings.enableTextToSpeech = enable;
        if (this.autoSave) this.saveSettings();
    }

    // 設定系統提示詞
    getSystemPrompt() {
        return this.settings.systemPrompt || this.defaultSettings.systemPrompt;
    }

    setSystemPrompt(prompt) {
        this.settings.systemPrompt = prompt;
        if (this.autoSave) this.saveSettings();
    }

    // 批次更新設定
    beginUpdate() {
        this.autoSave = false;
    }

    endUpdate() {
        this.autoSave = true;
        this.saveSettings();
    }
}

// 建立設定服務的單例實例
const settingsService = new SettingsService(); 

File: Services/SettingsService.cs
using ScreenAgent.Models;
using Newtonsoft.Json;
using System.IO;

namespace ScreenAgent.Services
{
    public class SettingsService
    {
        private readonly string _configPath;
        private AppSettings _settings;
        private bool _autoSave = true;

        public SettingsService(bool autoSave = true)
        {
            _autoSave = autoSave;
            _configPath = Path.Combine(
                Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData),
                "ScreenAgent",
                "settings.json");
            LoadSettings();
        }
        
        private void LoadSettings()
        {
            try
            {
                if (File.Exists(_configPath))
                {
                    var json = File.ReadAllText(_configPath);
                    _settings = JsonConvert.DeserializeObject<AppSettings>(json);
                }
                else
                {
                    _settings = new AppSettings
                    {
                        ApiKey = "",
                        ShowConversationHistory = true,
                        CaptureFrequencyInSeconds = 1
                    };
                    SaveSettings();
                }
            }
            catch
            {
                _settings = new AppSettings
                {
                    ApiKey = "",
                    ShowConversationHistory = true,
                    CaptureFrequencyInSeconds = 1
                };
            }
        }
        
        public void SaveSettings()
        {
            try
            {
                Directory.CreateDirectory(Path.GetDirectoryName(_configPath));
                var json = JsonConvert.SerializeObject(_settings, Formatting.Indented);
                File.WriteAllText(_configPath, json);
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error saving settings: {ex.Message}");
            }
        }
        
        public string GetApiKey() => _settings.ApiKey;
        
        public bool GetShowConversationHistory() => _settings.ShowConversationHistory;
        
        public int GetCaptureFrequencyInMilliseconds() => _settings.CaptureFrequencyInSeconds * 1000;
        
        public string GetTextToSpeechVoiceName() => _settings.TextToSpeechVoiceName;

        public string GetGeminiModel() => _settings.GeminiModel;

        public bool GetEnableTextToSpeech() => _settings.EnableTextToSpeech;
        
        // 語音識別相關 Getter
        public string GetSpeechLanguage() => _settings.SpeechLanguage;
        
        public bool GetEnableSpeechToText() => _settings.EnableSpeechToText;
        
        // 取得送出訊息時是否要截圖
        public bool GetCaptureOnSend() => _settings.CaptureOnSend;
        
        // 取得工具模式設定
        public ToolMode GetToolMode() => _settings.ToolMode;
        
        // 修改設定值的方法，增加不自動保存的選項
        private void SetValue<T>(Action<T> setter, T value)
        {
            setter(value);
            if (_autoSave)
            {
                SaveSettings();
            }
        }
        
        public void SetApiKey(string key) => SetValue(k => _settings.ApiKey = k, key);
        
        public void SetShowConversationHistory(bool show) => SetValue(s => _settings.ShowConversationHistory = s, show);
        
        public void SetCaptureFrequencyInSeconds(int seconds) => SetValue(s => _settings.CaptureFrequencyInSeconds = s, seconds);
        
        public void SetTextToSpeechVoiceName(string name) => SetValue(n => _settings.TextToSpeechVoiceName = n, name);
        
        public void SetGeminiModel(string model) => SetValue(m => _settings.GeminiModel = m, model);
        
        public void SetEnableTextToSpeech(bool enable) => SetValue(e => _settings.EnableTextToSpeech = e, enable);
        
        // 語音識別相關 Setter
        public void SetSpeechLanguage(string language) => SetValue(l => _settings.SpeechLanguage = l, language);
        
        public void SetEnableSpeechToText(bool enable) => SetValue(e => _settings.EnableSpeechToText = e, enable);
        
        // 設定送出訊息時是否要截圖
        public void SetCaptureOnSend(bool enable) => SetValue(e => _settings.CaptureOnSend = e, enable);
        
        // 設定工具模式
        public void SetToolMode(ToolMode mode) => SetValue(m => _settings.ToolMode = m, mode);
        
        // 批次更新設定而不是每次都保存
        public void BeginUpdate()
        {
            _autoSave = false;
        }
        
        public void EndUpdate()
        {
            _autoSave = true;
            SaveSettings();
        }

        public string GetSystemPrompt() => _settings.SystemPrompt;
        
        public void SetSystemPrompt(string prompt) => SetValue(p => _settings.SystemPrompt = p, prompt);
    }
}